dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/FourEyeApproverActionlet.java

```diff
@@ -0,0 +1,166 @@
+                            "Number of Approvers",
+                            String.valueOf(DEFAULT_MINIMUM_CONTENT_APPROVERS), true));
+            ACTIONLET_PARAMETERS
+                    .add(new WorkflowActionletParameter(PARAM_EMAIL_SUBJECT,
+                            "Email Subject",
+                            "'4 Eye' Approval Required", true));
+            ACTIONLET_PARAMETERS
+                    .add(new WorkflowActionletParameter(PARAM_EMAIL_BODY, "Email Message",
+                            null,
+                            false));
+        }
+        return ACTIONLET_PARAMETERS;
+    }
+
+    @Override
+    public String getName() {
+        return ACTIONLET_NAME;
+    }
+
+    @Override
+    public String getHowTo() {
+        return HOW_TO;
+    }
+
+    @Override
+    public boolean stopProcessing() {
+        return this.shouldStop;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof WorkFlowActionlet) {
+            return getClass().equals(obj.getClass());
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return (shouldStop ? 1 : 0);
+    }
+
+    @Override
+    public void executeAction(final WorkflowProcessor processor,
+            final Map<String, WorkflowActionClassParameter> params) {
+        final String userIds = getParameterValue(params.get(PARAM_CONTENT_APPROVERS));
+        final int minimumContentApprovers = ConversionUtils
+                .toInt(getParameterValue(params.get(PARAM_MINIMUM_CONTENT_APPROVERS)),
+                        DEFAULT_MINIMUM_CONTENT_APPROVERS);
+        final String emailSubject = getParameterValue(params.get(PARAM_EMAIL_SUBJECT));
+        final String emailBody = getParameterValue(params.get(PARAM_EMAIL_BODY));
+        final boolean isHtml = getParameterValue(params.get(PARAM_IS_HTML), false);
+        final Set<User> requiredContentApprovers = getUsersFromIds(userIds, USER_ID_DELIMITER);
+        // Add this approval to the history
+        final WorkflowHistory history = new WorkflowHistory();
+        history.setActionId(processor.getAction().getId());
+        history.setMadeBy(processor.getUser().getUserId());
+        List<WorkflowHistory> historyList = processor.getHistory();
+        if (null == historyList) {
+            historyList = new ArrayList<>();
+        }
+        historyList.add(history);
+        final Set<User> hasApproved = getApproversFromHistory(historyList, requiredContentApprovers,
+                processor.getAction().getId(), minimumContentApprovers);
+        if (hasApproved.size() < minimumContentApprovers) {
+            this.shouldStop = true;
+            // Keep the workflow process on the same step
+            processor.setNextStep(processor.getStep());
+            // Send email to users who have NOT approved only
+            final List<String> emails = new ArrayList<>();
```

Comment:
This could be extracted to a private method sendMailsToNotApprovingUsers(). it would include this for loop and the sending part below from lines 159-162